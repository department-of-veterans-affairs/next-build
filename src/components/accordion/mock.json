{
  "type": "paragraph--basic_accordion",
  "id": "ca7c4618-6701-4f84-a636-8209c05c9eca",
  "drupal_internal__id": 38313,
  "drupal_internal__revision_id": 284296,
  "langcode": "en",
  "status": true,
  "created": "2021-04-28T01:16:36+00:00",
  "parent_id": "16142",
  "parent_type": "node",
  "parent_field_name": "field_content_block",
  "behavior_settings": [],
  "default_langcode": true,
  "revision_translation_affected": true,
  "content_translation_source": "und",
  "content_translation_outdated": false,
  "content_translation_changed": null,
  "field_header": "What are Vet Centers?",
  "field_rich_wysiwyg": {
    "value": "<p>Vet Centers are small, non-medical, counseling centers conveniently located in your community. They're staffed by highly trained counselors and team members dedicated to seeing you through the challenges that come with managing life during and after the military.</p>\r\n\r\n<p>Whether you come in for one-on-one counseling or to participate in a group session, at Vet Centers you can form social connections, try new things, and build a support system with people who understand you and want to help you succeed.</p>\r\n",
    "format": "rich_text_limited",
    "processed": "<p>Vet Centers are small, non-medical, counseling centers conveniently located in your community. They're staffed by highly trained counselors and team members dedicated to seeing you through the challenges that come with managing life during and after the military.</p>\n<p>Whether you come in for one-on-one counseling or to participate in a group session, at Vet Centers you can form social connections, try new things, and build a support system with people who understand you and want to help you succeed.</p>\n"
  },
  "links": {
    "self": {
      "href": "https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/jsonapi/paragraph/basic_accordion/ca7c4618-6701-4f84-a636-8209c05c9eca?resourceVersion=id%3A284296"
    }
  },
  "paragraph_type": {
    "type": "paragraphs_type--paragraphs_type",
    "id": "5f3ba120-da59-4c9a-b7d8-5fd72bf3eecd",
    "resourceIdObjMeta": {
      "drupal_internal__target_id": "basic_accordion"
    }
  },
  "relationshipNames": ["paragraph_type"]
}
