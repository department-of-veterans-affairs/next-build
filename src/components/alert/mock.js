export const mockResponse = {
  type: 'paragraph--alert',
  id: '019070f3-9e3e-4c69-83e4-c7eace8c414f',
  drupal_internal__id: 6003,
  drupal_internal__revision_id: 27807,
  langcode: 'en',
  status: true,
  created: '2019-09-04T13:45:29+00:00',
  parent_id: '876',
  parent_type: 'node',
  parent_field_name: 'field_content_block',
  behavior_settings: [],
  default_langcode: true,
  revision_translation_affected: true,
  content_translation_source: 'und',
  content_translation_outdated: false,
  content_translation_changed: null,
  field_alert_heading: 'Blue Water Navy Veterans and Agent Orange',
  field_alert_type: 'information',
  links: {
    self: {
      href: 'https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/jsonapi/paragraph/alert/019070f3-9e3e-4c69-83e4-c7eace8c414f?resourceVersion=id%3A27807',
    },
  },
  paragraph_type: {
    type: 'paragraphs_type--paragraphs_type',
    id: 'daa6bb41-1b12-46f7-a355-96ceb6fa18dc',
    resourceIdObjMeta: {
      drupal_internal__target_id: 'alert',
    },
  },
  field_alert_block_reference: null,
  field_va_paragraphs: [
    {
      type: 'paragraph--wysiwyg',
      id: 'abb45dcb-b33f-4080-b20f-2f89d4281806',
      drupal_internal__id: 6004,
      drupal_internal__revision_id: 27808,
      langcode: 'en',
      status: true,
      created: '2019-09-04T13:45:29+00:00',
      parent_id: '6003',
      parent_type: 'paragraph',
      parent_field_name: 'field_va_paragraphs',
      behavior_settings: [],
      default_langcode: true,
      revision_translation_affected: true,
      content_translation_source: 'und',
      content_translation_outdated: false,
      content_translation_changed: null,
      field_wysiwyg: {
        value:
          '<div class="usa-alert-body" ><p>Blue Water Navy Veterans are now entitled to a presumption of service connection for illnesses related to Agent Orange exposure. This is a result of <a href="https://www.congress.gov/bill/116th-congress/house-bill/299/text" target="_blank" rel="noopener ">Public Law 116-23</a>, the Blue Water Navy Vietnam Veterans Act of 2019. The law was signed on June 25, 2019, and takes effect on January 1, 2020.</p><p>Read below to learn more about eligibility requirements and how to file a claim. If we denied your claim in the past, you can file a new claim based on this change in law.</p></div>',
        format: 'rich_text',
        processed:
          '<p>Blue Water Navy Veterans are now entitled to a presumption of service connection for illnesses related to Agent Orange exposure. This is a result of <a href="https://www.congress.gov/bill/116th-congress/house-bill/299/text" target="_blank">Public Law 116-23</a>, the Blue Water Navy Vietnam Veterans Act of 2019. The law was signed on June 25, 2019, and takes effect on January 1, 2020.</p><p>Read below to learn more about eligibility requirements and how to file a claim. If we denied your claim in the past, you can file a new claim based on this change in law.</p>',
      },
      links: {
        self: {
          href: 'https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/jsonapi/paragraph/wysiwyg/abb45dcb-b33f-4080-b20f-2f89d4281806?resourceVersion=id%3A27808',
        },
      },
      resourceIdObjMeta: {
        target_revision_id: 27808,
        drupal_internal__target_id: 6004,
      },
      paragraph_type: {
        type: 'paragraphs_type--paragraphs_type',
        id: '885e4b61-cfd2-44b9-94ae-f068ba2b48b6',
        resourceIdObjMeta: {
          drupal_internal__target_id: 'wysiwyg',
        },
      },
      relationshipNames: ['paragraph_type'],
    },
  ],
  relationshipNames: [
    'paragraph_type',
    'field_alert_block_reference',
    'field_va_paragraphs',
  ],
}
