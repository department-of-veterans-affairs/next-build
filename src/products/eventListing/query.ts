import { QueryData, QueryFormatter, QueryParams } from 'next-drupal-query'
import { queries } from '@/data/queries'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'
import { NodeEvent, NodeEventListing } from '@/types/drupal/node'
import { Menu } from '@/types/drupal/menu'
import { EventListing } from '@/products/eventListing/formatted-type'
import { RESOURCE_TYPES } from '@/lib/constants/resourceTypes'
import { PAGE_SIZES } from '@/lib/constants/pageSizes'
import { ExpandedStaticPropsContext } from '@/lib/drupal/staticProps'
import {
  entityBaseFields,
  fetchAndConcatAllResourceCollectionPages,
  fetchSingleEntityOrPreview,
  getMenu,
} from '@/lib/drupal/query'
import { buildSideNavDataFromMenu } from '@/lib/drupal/facilitySideNav'

const PAGE_SIZE = PAGE_SIZES[RESOURCE_TYPES.EVENT_LISTING]
// See listingParams for more information about this value and its use.
const OUTREACH_CAL_UUID = '30295cab-89be-4173-a33c-8c0ca0a85d07'

// Define the query params for fetching node--event_listing.
export const params: QueryParams<null> = () => {
  return new DrupalJsonApiParams().addInclude(['field_office'])
}

// Define the option types for the data loader.
export type EventListingDataOpts = {
  id: string
  context?: ExpandedStaticPropsContext
}

type EventListingData = {
  entity: NodeEventListing
  events: NodeEvent[]
  menu?: Menu
  totalItems: number
  totalPages: number
}

const listingParams: QueryParams<string> = (listingEntityId: string) => {
  let eventParams = queries
    .getParams(`${RESOURCE_TYPES.EVENT}--teaser`)
    .addGroup('outreach_cal_group', 'OR')
    .addFilter('field_listing.id', listingEntityId, '=', 'outreach_cal_group')
    .addSort('-created')
  // The National Outreach Calendar is a special case.
  // https://www.va.gov/outreach-and-events/events/
  // There are two ways an event can end up on this calendar:
  // - directly assigned on field_listing (the way most events do)
  // - assigned via the checkbox field field_publish_to_outreach_cal
  // This additional filter grabs that second case.
  if (listingEntityId === OUTREACH_CAL_UUID) {
    eventParams = eventParams.addFilter(
      'field_publish_to_outreach_cal',
      '1',
      '=',
      'outreach_cal_group'
    )
  }
  return eventParams
}

// Implement the data loader.
export const data: QueryData<EventListingDataOpts, EventListingData> = async (
  opts
) => {
  const entity = (await fetchSingleEntityOrPreview(
    opts,
    RESOURCE_TYPES.EVENT_LISTING,
    params
  )) as NodeEventListing

  // Fetch list of events related to this listing
  const { data: events } =
    await fetchAndConcatAllResourceCollectionPages<NodeEvent>(
      RESOURCE_TYPES.EVENT,
      listingParams(entity.id),
      PAGE_SIZE
    )

  // Fetch the menu name dynamically off of the field_office reference if available.
  // The `/outreach-and-events/events` event listing page has no menu attached to it.
  let menu = null
  if (entity.field_office.field_system_menu) {
    menu = await getMenu(
      entity.field_office.field_system_menu.resourceIdObjMeta
        .drupal_internal__target_id
    )
  }

  return {
    entity,
    events,
    menu,
    totalItems: events.length,
    totalPages: 1, // We don't want to paginate event listing pages. The widget handles pagination.
  }
}

export const formatter: QueryFormatter<EventListingData, EventListing> = ({
  entity,
  events,
  menu,
  totalItems,
  totalPages,
}) => {
  const formattedEvents = events.map((event) => {
    return queries.formatData(`${RESOURCE_TYPES.EVENT}--teaser`, event)
  })

  let formattedMenu = null
  if (menu !== null)
    formattedMenu = buildSideNavDataFromMenu(entity.path.alias, menu)

  return {
    ...entityBaseFields(entity),
    introText: entity.field_intro_text,
    events: formattedEvents,
    menu: formattedMenu,
    totalItems,
    totalPages,
  }
}
