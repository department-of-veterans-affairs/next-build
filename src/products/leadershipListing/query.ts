import { QueryData, QueryFormatter, QueryParams } from 'next-drupal-query'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'
import { NodeLeadershipListing } from '@/types/drupal/node'
import { LeadershipListing } from '@/products/leadershipListing/formatted-type'
import { RESOURCE_TYPES } from '@/lib/constants/resourceTypes'
import { ExpandedStaticPropsContext } from '@/lib/drupal/staticProps'
import { buildSideNavDataFromMenu } from '@/lib/drupal/facilitySideNav'
import { formatter as formatImage } from '@/data/queries/mediaImage'
import { formatter as formatPhone } from '@/data/queries/phoneNumber'
import { formatter as formatAdministration } from '@/data/queries/administration'
import {
  getLovellVariantOfUrl,
  getOppositeChildVariant,
  getLovellVariantOfBreadcrumbs,
} from '@/lib/drupal/lovell/utils'

import {
  entityBaseFields,
  fetchSingleEntityOrPreview,
  getMenu,
} from '@/lib/drupal/query'
import { Menu } from '@/types/drupal/menu'

// Define the query params for fetching node--leadership_listing.
export const params: QueryParams<null> = () => {
  return new DrupalJsonApiParams().addInclude([
    'field_leadership',
    'field_office',
    'field_leadership.field_media.image',
    'field_leadership.field_telephone',
    'field_leadership.field_office',
    'field_administration',
  ])
}

// Define the option types for the data loader.
export type LeadershipListingDataOpts = {
  id: string
  context?: ExpandedStaticPropsContext
}

export type LeadershipListingData = {
  entity: NodeLeadershipListing
  menu: Menu | null
  lovell?: ExpandedStaticPropsContext['lovell']
}

// Implement the data loader.
export const data: QueryData<
  LeadershipListingDataOpts,
  LeadershipListingData
> = async (opts) => {
  const entity = (await fetchSingleEntityOrPreview(
    opts,
    RESOURCE_TYPES.LEADERSHIP_LISTING,
    params
  )) as NodeLeadershipListing

  const menu = entity.field_office
    ? await getMenu(
        entity.field_office.field_system_menu.resourceIdObjMeta
          .drupal_internal__target_id
      )
    : null

  return { entity, menu, lovell: opts.context?.lovell }
}

export const formatter: QueryFormatter<
  LeadershipListingData,
  LeadershipListing
> = ({ entity, menu, lovell }) => {
  let { breadcrumbs } = entity
  if (lovell?.isLovellVariantPage) {
    breadcrumbs = getLovellVariantOfBreadcrumbs(breadcrumbs, lovell.variant)
  }
  const formattedMenu =
    menu !== null ? buildSideNavDataFromMenu(entity.path.alias, menu) : null
  const formattedProfiles = entity.field_leadership
    .filter((profile) => profile.status === true)
    .map((profile) => {
      const formattedLink = () => {
        if (!profile.path?.alias || !profile.field_complete_biography_create) {
          return ''
        }
        if (lovell?.isLovellVariantPage) {
          return getLovellVariantOfUrl(profile.path.alias, lovell.variant)
        }
        return profile.path.alias
      }
      return {
        firstName: profile.field_name_first || '',
        lastName: profile.field_last_name || '',
        vamcTitle: profile.field_office?.title || '',
        description: profile.field_description || '',
        suffix: profile.field_suffix || '',
        phoneNumber: formatPhone(profile.field_telephone),
        media: formatImage(profile.field_media),
        link: formattedLink(),
        id: profile.id,
      }
    })
  return {
    ...entityBaseFields(entity),
    breadcrumbs,
    introText: entity.field_intro_text,
    profiles: formattedProfiles,
    menu: formattedMenu,
    administration: formatAdministration(entity.field_administration),
    lovellVariant: lovell?.variant ?? null,
    lovellSwitchPath: lovell?.isLovellVariantPage
      ? getLovellVariantOfUrl(
          entity.path.alias,
          getOppositeChildVariant(lovell?.variant)
        )
      : null,
  }
}
