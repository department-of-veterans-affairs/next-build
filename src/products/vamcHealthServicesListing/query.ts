import { QueryData, QueryFormatter, QueryParams } from 'next-drupal-query'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'
import { NodeVamcHealthServicesListing } from '@/types/drupal/node'
import { VamcHealthServicesListing } from '@/products/vamcHealthServicesListing/formatted-type'
import { RESOURCE_TYPES } from '@/lib/constants/resourceTypes'
import { ExpandedStaticPropsContext } from '@/lib/drupal/staticProps'
import {
  entityBaseFields,
  fetchSingleEntityOrPreview,
} from '@/lib/drupal/query'

// Define the query params for fetching node--health_services_listing.
export const params: QueryParams<null> = () => {
  return new DrupalJsonApiParams()
}

// Define the option types for the data loader.
export type VamcHealthServicesListingDataOpts = {
  id: string
  context?: ExpandedStaticPropsContext
}

// Implement the data loader.
export const data: QueryData<
  VamcHealthServicesListingDataOpts,
  NodeVamcHealthServicesListing
> = async (opts): Promise<NodeVamcHealthServicesListing> => {
  const entity = (await fetchSingleEntityOrPreview(
    opts,
    RESOURCE_TYPES.VAMC_HEALTH_SERVICES_LISTING,
    params
  )) as NodeVamcHealthServicesListing

  return entity
}

export const formatter: QueryFormatter<
  NodeVamcHealthServicesListing,
  VamcHealthServicesListing
> = (entity: NodeVamcHealthServicesListing) => {
  return {
    ...entityBaseFields(entity),
    introText: entity.field_intro_text,
  }
}
