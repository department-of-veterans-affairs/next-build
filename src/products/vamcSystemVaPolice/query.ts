import { QueryData, QueryFormatter, QueryParams } from 'next-drupal-query'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'
import { NodeVamcSystemVaPolice } from '@/types/drupal/node'
import { VamcSystemVaPolice } from '@/products/vamcSystemVaPolice/formatted-type'
import { RESOURCE_TYPES } from '@/lib/constants/resourceTypes'
import { ExpandedStaticPropsContext } from '@/lib/drupal/staticProps'
import {
  entityBaseFields,
  fetchSingleEntityOrPreview,
  getMenu,
} from '@/lib/drupal/query'
import { Menu } from '@/types/drupal/menu'
import { buildSideNavDataFromMenu } from '@/lib/drupal/facilitySideNav'
import { getHtmlFromField } from '@/lib/utils/getHtmlFromField'
import { getLovellVariantOfBreadcrumbs } from '@/lib/drupal/lovell/utils'
import {
  ParagraphCCVetCenterFaqs,
  ParagraphQaSection,
} from '@/types/drupal/paragraph'
import {
  entityFetchedParagraphsToNormalParagraphs,
  formatParagraph,
} from '@/lib/drupal/paragraphs'
import { QaSection } from '@/types/formatted/qaSection'

// Define the query params for fetching node--vamc_system_va_police.
export const params: QueryParams<null> = () => {
  return new DrupalJsonApiParams().addInclude([
    'field_administration',
    'field_office',
    'field_phone_numbers_paragraph',
  ])
}

// Define the option types for the data loader.
export type VamcSystemVaPoliceDataOpts = {
  id: string
  context?: ExpandedStaticPropsContext
}

type VamcSystemVaPoliceData = {
  entity: NodeVamcSystemVaPolice
  menu: Menu | null
  lovell?: ExpandedStaticPropsContext['lovell']
}

// Implement the data loader.
export const data: QueryData<
  VamcSystemVaPoliceDataOpts,
  VamcSystemVaPoliceData
> = async (opts): Promise<VamcSystemVaPoliceData> => {
  const entity = (await fetchSingleEntityOrPreview(
    opts,
    RESOURCE_TYPES.VAMC_SYSTEM_VA_POLICE,
    params
  )) as NodeVamcSystemVaPolice

  if (!entity) {
    throw new Error(
      `NodeVamcSystemVaPolice entity not found for id: ${opts.id}`
    )
  }

  // Fetch the menu name dynamically off of the field_office reference
  const menu = entity.field_office.field_system_menu
    ? await getMenu(
        entity.field_office.field_system_menu.resourceIdObjMeta
          ?.drupal_internal__target_id
      )
    : null

  return { entity, menu, lovell: opts.context?.lovell }
}

// Similarly, this formats centralized content FAQs to match what our QA components are expecting
const formatFaq = (faqs: ParagraphCCVetCenterFaqs) => {
  const normalizedQaSection = entityFetchedParagraphsToNormalParagraphs({
    type: faqs.target_type,
    bundle: faqs.fetched_bundle,
    ...faqs.fetched,
  }) as ParagraphQaSection
  return formatParagraph(normalizedQaSection) as QaSection
}

export const formatter: QueryFormatter<
  VamcSystemVaPoliceData,
  VamcSystemVaPolice
> = ({ entity, menu, lovell }) => {
  let { breadcrumbs } = entity
  if (lovell?.isLovellVariantPage) {
    breadcrumbs = getLovellVariantOfBreadcrumbs(breadcrumbs, lovell.variant)
  }
  const formattedMenu = buildSideNavDataFromMenu(entity.path.alias, menu)

  return {
    ...entityBaseFields(entity),
    title: entity.title,
    breadcrumbs,
    path: entity.path.alias,
    menu: formattedMenu,
    policeOverview: {
      type: 'paragraph--wysiwyg',
      id: entity.field_cc_va_police_overview.target_id ?? '',
      html:
        getHtmlFromField(
          entity.field_cc_va_police_overview?.fetched?.field_wysiwyg?.[0]
        ) || '',
    },
    system: entity.field_office?.title || '',
    phoneNumber: {
      extension:
        entity.field_phone_numbers_paragraph?.[0]?.field_phone_extension || '',
      number:
        entity.field_phone_numbers_paragraph?.[0]?.field_phone_number || '',
      phoneType:
        entity.field_phone_numbers_paragraph?.[0]?.field_phone_number_type ||
        '',
      id: entity.field_phone_numbers_paragraph?.[0]?.id ?? '',
      type: 'paragraph--phone_number',
    },
    policeReport: {
      id: entity.field_cc_police_report?.[0]?.target_id ?? '',
      type: 'paragraph--featured_content',
      title:
        entity.field_cc_police_report?.fetched?.field_section_header?.[0]
          ?.value || '',
      description:
        entity.field_cc_police_report?.fetched?.field_description?.[0]
          ?.processed || '',
      link: {
        id: entity.field_cc_police_report?.fetched?.field_cta?.[0]?.id ?? '',
        type: 'paragraph--button',
        label:
          entity.field_cc_police_report?.fetched?.field_cta?.[0]
            ?.field_button_label?.[0]?.value || '',
        url:
          entity.field_cc_police_report?.fetched?.field_cta?.[0]
            ?.field_button_link?.[0]?.uri || '',
      },
    },
    faqs: formatFaq(entity.field_cc_faq),
  }
}
