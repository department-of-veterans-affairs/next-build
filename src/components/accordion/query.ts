import { QueryFormatter } from 'next-drupal-query'
import { ParagraphAccordion } from '@/types/drupal/paragraph'
import { AccordionItem } from '@/components/accordion/formatted-type'
import { getHtmlFromField } from '@/lib/utils/getHtmlFromField'

export const formatter: QueryFormatter<ParagraphAccordion, AccordionItem> = (
  entity: ParagraphAccordion
) => {
  return {
    type: entity.type as AccordionItem['type'],
    id: entity.id,
    entityId: entity.drupal_internal__id,
    header: entity.field_header || null,
    html: getHtmlFromField(entity.field_rich_wysiwyg),
  }
}
