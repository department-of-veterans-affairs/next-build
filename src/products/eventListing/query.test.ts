/**
 * @jest-environment node
 */

import { NodeEvent, NodeEventListing } from '@/types/drupal/node'
import { queries } from '@/data/queries'
import { mockResponse } from '@/products/eventListing/mock.js'
import mockEventData from '@/mocks/event.mock.json'
import { RESOURCE_TYPES } from '@/lib/constants/resourceTypes'
import { params } from './query'

const EventListingMock: NodeEventListing = mockResponse
const EventMock: NodeEvent[] = [mockEventData]

describe('DrupalJsonApiParams configuration', () => {
  test('params function sets the correct include fields', () => {
    const paramsInstance = params()
    const queryString = decodeURIComponent(paramsInstance.getQueryString())
    expect(queryString).toMatch(/include=field_office/)
  })
})

describe(`${RESOURCE_TYPES.EVENT_LISTING} formatData`, () => {
  test('outputs formatted data', () => {
    expect(
      queries.formatData(RESOURCE_TYPES.EVENT_LISTING, {
        entity: EventListingMock,
        events: EventMock,
        menu: { items: [], tree: [] },
        totalItems: EventMock.length,
        totalPages: 1,
      })
    ).toMatchSnapshot()
  })
})
