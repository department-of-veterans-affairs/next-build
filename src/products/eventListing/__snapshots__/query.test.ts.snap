// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`node--event_listing formatData outputs formatted data 1`] = `
{
  "breadcrumbs": [
    {
      "options": [],
      "title": "Home",
      "uri": "https://va-gov-cms.ddev.site/",
    },
    {
      "options": [],
      "title": "VA Wilkes-Barre health care",
      "uri": "https://va-gov-cms.ddev.site/wilkes-barre-health-care",
    },
    {
      "options": [],
      "title": "News and events",
      "uri": "internal:#",
    },
    {
      "options": [],
      "title": "Events",
      "uri": "internal:#",
    },
  ],
  "entityId": 3036,
  "entityPath": "/wilkes-barre-health-care/events",
  "events": [
    {
      "changed": "2023-08-16T15:20:30+00:00",
      "entityBundle": "node--event",
      "entityId": "16349f16-be65-46e3-9660-1d3d598a4a0b",
      "entityPublished": true,
      "entityUrl": {
        "path": "/central-iowa-health-care/events/52265",
      },
      "fieldAdditionalInformationAbo": null,
      "fieldAdditionalListings": null,
      "fieldAddress": {
        "addressLine1": "9000 Douglas Ave",
        "addressLine2": null,
        "administrativeArea": "IA",
        "countryCode": "US",
        "locality": "Urbandale",
        "postalCode": null,
      },
      "fieldAdministration": {
        "entity": {
          "entityId": "3a530a55-e922-48de-b18c-9c9f92a9ae98",
        },
      },
      "fieldBody": {
        "format": "rich_text",
        "processed": "<p>Pickleball Club</p>

<p>Meets Thursdays from 9 to 11 a.m.Â </p>

<p>Contact Kay Queck (515) 214-4578</p>",
        "value": "<p>Pickleball Club</p>

<p>Meets Thursdays from 9 to 11 a.m.&nbsp;</p>

<p>Contact Kay Queck (515) 214-4578</p>
",
      },
      "fieldCtaEmail": "test@va.gov",
      "fieldDatetimeRangeTimezone": [
        {
          "duration": 120,
          "endTime": "2023-09-07T16:00:00+00:00",
          "endValue": 1694102400,
          "end_value": "2023-09-07T16:00:00+00:00",
          "rrule": 180,
          "rrule_index": 1,
          "startTime": "2023-09-07T14:00:00+00:00",
          "timezone": "America/Chicago",
          "value": 1694095200,
        },
        {
          "duration": 120,
          "endTime": "2023-09-14T16:00:00+00:00",
          "endValue": 1694707200,
          "end_value": "2023-09-14T16:00:00+00:00",
          "rrule": 180,
          "rrule_index": 2,
          "startTime": "2023-09-14T14:00:00+00:00",
          "timezone": "America/Chicago",
          "value": 1694700000,
        },
      ],
      "fieldDescription": "Pickleball ",
      "fieldEventCost": "Free",
      "fieldEventCta": null,
      "fieldEventRegistrationrequired": false,
      "fieldFacilityLocation": null,
      "fieldFeatured": false,
      "fieldHowToSignUp": "email",
      "fieldLink": null,
      "fieldListing": {
        "entity": {
          "entityId": "55cb823c-df6d-4f77-814d-6024e08cd4f6",
        },
      },
      "fieldLocationHumanreadable": "Walker Johnston Park",
      "fieldLocationType": "non_facility",
      "fieldOrder": null,
      "fieldUrlOfAnOnlineEvent": null,
      "title": "Pickleball Club",
    },
  ],
  "id": "4c24a406-37dc-4709-9ba7-24f430973cfb",
  "introText": "Learn more about events in our VA Wilkes-Barre health care community, including classes on health and wellness.",
  "lastUpdated": "2021-02-11T16:15:01+00:00",
  "menu": {
    "data": null,
    "rootPath": "/wilkes-barre-health-care/events/",
  },
  "metatags": [
    {
      "attributes": {
        "content": "Events | VA Wilkes-Barre health care | Veterans Affairs",
        "name": "title",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "Learn more about events in our VA Wilkes-Barre health care community, including classes on health and wellness.",
        "name": "description",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "href": "https://www.va.gov/img/design/logo/va-og-image.png",
        "rel": "image_src",
      },
      "tag": "link",
    },
    {
      "attributes": {
        "content": "Veterans Affairs",
        "property": "og:site_name",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "Events | Veterans Affairs",
        "property": "og:title",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "Learn more about events in our VA Wilkes-Barre health care community, including classes on health and wellness.",
        "property": "og:description",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "https://www.va.gov/img/design/logo/va-og-image.png",
        "property": "og:image",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "U.S. Department of Veterans Affairs",
        "property": "og:image:alt",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "summary_large_image",
        "name": "twitter:card",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "Learn more about events in our VA Wilkes-Barre health care community, including classes on health and wellness.",
        "name": "twitter:description",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "Events | VA Wilkes-Barre health care | Veterans Affairs",
        "name": "twitter:title",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "@DeptVetAffairs",
        "name": "twitter:site",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "https://www.va.gov/img/design/logo/va-og-image.png",
        "name": "twitter:image",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "U.S. Department of Veterans Affairs",
        "name": "twitter:image:alt",
      },
      "tag": "meta",
    },
  ],
  "moderationState": "published",
  "published": true,
  "title": "Events",
  "totalItems": 1,
  "totalPages": 1,
  "type": "node--event_listing",
}
`;
